packages/agent/src/agent/http/index.ts

```diff
@@ -293,10 +293,10 @@ export class HttpAgent implements Agent {
       request: {
         body: null,
         method: 'POST',
-        headers: {
+        headers: new Headers({
           'Content-Type': 'application/cbor',
           ...(this._credentials ? { Authorization: 'Basic ' + btoa(this._credentials) } : {}),
```

Comment:
`btoa` is deprecated in node. would this ever run in a node context? if it were, better to detect the environment and use `Buffer` instead

```typescript
if (isBrowserEnv()) {
  window.btoa(this._credentials);
}

if (isNodeEnv()) {
  Buffer.from(this._credentials).toString('base64')
}
```